src/transformers/pipelines/__init__.py

```diff
@@ -1074,6 +1100,32 @@ def pipeline(
                     if not is_pyctcdecode_available():
                         logger.warning("Try to install `pyctcdecode`: `pip install pyctcdecode")
 
+    if load_processor:
+        # Try to infer processor from model or config name (if provided as str)
+        if processor is None:
+            if isinstance(model_name, str):
+                processor = model_name
+            elif isinstance(config, str):
+                processor = config
+            else:
+                # Impossible to guess what is the right processor here
+                raise Exception(
+                    "Impossible to guess which processor to use. "
+                    "Please provide a processor instance or a path/identifier "
+                    "to a processor."
+                )
+
+        # Instantiate processor if needed
+        if isinstance(processor, (str, tuple)):
+            processor = AutoProcessor.from_pretrained(processor, _from_pipeline=task, **hub_kwargs, **model_kwargs)
+            if not isinstance(processor, ProcessorMixin):
+                warnings.warn(
+                    f"Processor will be not loaded, because {processor} is not an instance of `ProcessorMixin`. "
+                    f"Got type `{type(processor)}` instead.",
+                    UserWarning,
+                )
```

Comment:
Thanks for the questions!  
There was a discussion [here](https://github.com/huggingface/transformers/pull/32514#discussion_r1717215386). The main idea is that `AutoProcessor` can load not only processors but also basic processing classes, like `tokenizer`. 

Indeed, a misconfiguration in the model-pipeline-processor setup could trigger this and raising a warning + dropping the `processor` might be sufficient **only when the `processor` isn't needed in the `pipeline` at all.** 

However, with granular control, such a case shouldn't occur. Therefore, it seems more appropriate to replace the warning with an error to clearly indicate a misconfiguration. Otherwise, the error will happen later with a less clear message because the `processor` is None.

Reply:
Changed to error raise here https://github.com/huggingface/transformers/pull/32514/commits/bcce4dceb19b39225b11acbcc495a754fa6c802d